### ISSUE TITLE
Multiple container access to ScaleIO volume

### ISSUE THREAD

**Issue by vincenzodnp** [thecodeteam/rexray#1171]
# Summary
Hello TheCodeTeam, first of all thanks for your work!
Maybe one stupid question.
I have read the documentation about RexRay and Docker Swarm integration and currently I'm using a Docker Swarm cluster (4 nodes) with RexRay plugin and a ScaleIO cluster (3 nodes).

From the docs, it seems that I should be able to attach one volume at a time per host and, for the same reason, at one container at a time.
Just for fun, I tried to start a docker service with --replicas 3 and with a volume attached. What I noticed is that docker creates 3 container on the same host (that's what I was expecting) but granting them RW access to the volume.

Is it something wrong, or is it a desired state? Is it something making sense or absolutely useless?

Thanks

## Version
docker.io/rexray/scaleio:latest

## Steps To Reproduce
`docker service create --name redis-scaleio --replicas 3 --constraint 'node.role == worker' --mount type=volume,source=testvolume2,destination=/data -p 6379:6379 redis:alpine redis-server --appendonly yes`

**Comment by clintkitson** [comment]
Yes, this is intended Docker behavior. The ScaleIO volume gets attached once and then subsequent containers that are launched all get access to this one volume.

**Comment by vincenzodnp** [comment]
Thanks @clintkitson . So it is a sort of "data volume". Isn't it?
Are there any issues of using it in RW mode?
Thanks again

**Comment by clintkitson** [comment]
It depends on how the application itself is using that data. If the app knows to be writing to different directories, then it is aok.

A volume mounted to multiple containers is perfectly fine. In this case the kernel manages the R/W access to the filesystem as if the app was just two separate processes.

**Comment by vincenzodnp** [comment]
Ok. So the only constrain is ScaleIO volume mounted on *one* host at a time regardless the number of containers using it.
Thanks @clintkitson 
ps: is on k8s the same behavior? (I mean multiple pods mounting same ScaleIO PV volume?)

**Comment by vincenzodnp** [comment]
@clintkitson I tested the same behavior on Openshift (kubernetes). What I noticed different, is the behavior when you delete or scale up a deployment. On docker swarm, rexray forces docker to re-schedule all replicas on the same other host. On K8s, it's random. If the schedurler starts pod on the same host where the first pod is running, no issue at all. If scheduler starts the pod on other host rather the one where first pod is running, I get errors (and it's right!).
May I suppose the rexray functionalities were not migrated into the build-in ScaleIO k8s plugin?
Thanks.

ps: if this question is not the right place, please feel free to tell me ;)

**Comment by clintkitson** [comment]
That's correct. This is a limitation in Kubernetes knowledge regarding scheduling based on plugin's capabilities. It currently does not include storage in its intelligence for scheduling. 

Even though ScaleIO can mount a volume to multiple hosts at the same time, you wouldn't do that for most workloads. This requires a clustered file system to work properly, ie. you can't mount an EXT4 filesystem to multiple hosts in any way.
